components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -178,6 +180,25 @@ void AIChatTabHelper::OnTabContentRetrieved(int64_t for_navigation_id,
     VLOG(1) << __func__ << ": No data";
     return;
   }
+
+  // Parse XML
```

Comment:
IMO all this parsing code should go in `page_content_fetcher.cc`, specificially the anonymous class `PageContentFetcher`, because that's what has the most knowledge about what type of data is coming back from the renderer, e.g. YouTube XML, WebVTT or x. Here you're passing a simple `parse_xml` flag, but there's no info about the XML schema, and we're just inferring it's YouTube.

Reply:
Right, `parse_xml` could better be named `is_youtube`.  I put the parsing logic in ai_chat_tab_helper.cc partly because I thought there may be a case for different prompts based on the video type (similar to is_video).  But since we treat all videos the same now, putting in page_content_fetcher.cc sounds good to me.

Reply:
Fair enough, but this file is getting pretty long and complicated and since we're removing most functionality from it via https://github.com/brave/brave-browser/issues/31623, including the prompt building, I think we can put this new piece somewhere else and reevaluate what extra info the prompt building needs to be given after the refactor.

Reply:
is this following the rule of 2?

Reply:
I believe so if we consider youtube a trustworthy source.  I will open a sec review

Reply:
i don't think we consider that a trustworthy source especially since some of this data might be controllable by the video creators. @nvonpentz i think this needs to be done in a sandboxed process or memory safe language.

Reply:
Okay, I opened https://github.com/brave/reviews/issues/1361 to track